lib/enmeshed/connector_api.rb

```diff
@@ -0,0 +1,144 @@
+
+      response = Net::HTTP.get(URI("#{@base_url}/api/v2/Account/IdentityInfo"), {'X-API-KEY' => @api_key})
+      @enmeshed_address = JSON.parse(response)['result']['address']
+
+      @enmeshed_address
+    end
+
+    def create_relationship_template(display_name, required_attributes)
+      display_name_id = create_display_name(display_name)
+
+      uri = URI("#{@base_url}/api/v2/RelationshipTemplates/Own")
+      response = Net::HTTP.post(uri, relationship_template_json(display_name, display_name_id, required_attributes), default_headers)
+
+      JSON.parse(response.body)['result']['id']
+    end
+
+    def create_display_name(display_name)
+      uri = URI("#{@base_url}/api/v2/Attributes")
+      response = Net::HTTP.post(uri, {
+        content: display_name_attribute_json(display_name),
+      }.to_json, default_headers)
+
+      JSON.parse(response.body)['result']['id']
+    end
+
+    def template_qr_code(template_id)
+      uri = URI("#{@base_url}/api/v2/RelationshipTemplates/#{template_id}")
+      Net::HTTP.get(uri, default_headers.merge(accept: 'image/png'))
+    end
+
+    def pending_relationships_for_rlt(template_id)
+      relationships_for_rlt(template_id, 'Pending')
+    end
+
+    def relationships_for_rlt(template_id, status = nil)
+      uri_string = "#{@base_url}/api/v2/Relationships?template.id=#{template_id}"
+      uri_string += "&status=#{status}" if status.present?
+      JSON.parse(Net::HTTP.get(URI(uri_string), default_headers))['result']
+    end
+
+    def accept_relationship_change(relationship_id, change_id)
+      respond_to_rel_change(relationship_id, change_id, 'Accept')
+    end
+
+    def reject_relationship_change(relationship_id, change_id)
+      respond_to_rel_change(relationship_id, change_id, 'Reject')
+    end
+
+    def respond_to_rel_change(relationship_id, change_id, action = 'Accept')
+      url = URI.parse("#{@base_url}/api/v2/Relationships/#{relationship_id}/Changes/#{change_id}/#{action}")
+      http = Net::HTTP.new(url.host, url.port)
+      request = Net::HTTP::Put.new(url.request_uri, default_headers)
+      request.body = '{"content": {}}'
+
+      http.request(request).code == '200'
+    end
+
+    def parse_userdata_in_rel_change(relship_change_json) # rubocop:disable Metrics/AbcSize
+      change_response_items = relship_change_json['request']['content']['response']['items'].pluck('items').flatten
+
+      read_attribute_accept_response_items = change_response_items.select {|item| item['@type'] == 'ReadAttributeAcceptResponseItem' }
+
+      userdata = {}
+
+      read_attribute_accept_response_items.each do |item|
+        attr_type = item['attribute']['value']['@type']
+        attr_value = item['attribute']['value']['value']
+
+        userdata[attr_type] = attr_value
+      end
+      userdata
+    end
+
+    def relationship_template_json(display_name, display_name_id, required_attributes) # rubocop:disable Metrics/MethodLength
```

Comment:
Thanks for updating. I think, it looks much better and consistent now. Just one further remark: Generally speaking, I usually try to avoid including an object type in a method or variable name (doesn't work always, but often). [Here](https://ruby-for-beginners.rubymonstas.org/bonus/good_names.html) are some further examples. Do you think we can just go with something like `relationship_template` or `display_name_attribute`?